### PR TITLE
Fix doxygen and source comments in `src/`

### PR DESCRIPTION
**Pull request by luzpaz**
## Main changes of this PR

Fix doxygen and source comments in `src/` directory  
Found via `codespell -q 3 -L asend,datas,precice,pullrequest -S CHANGELOG.md,./thirdparty,./docs/changelog`

## Motivation and additional information

self-explanatory

## Author's checklist

* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I ran `make format` to ensure everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [x] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

File: src/com/SocketCommunication.cpp [hunk 1]
```diff
@@ -912,13 +912,13 @@ std::string SocketCommunication::getIpAddress()
   auto pos = std::find_if(interfaces.begin(), interfaces.end(),
                           [&](Interface const &interface) { return interface.name == _networkName; });
   if (pos == interfaces.end()) {
-    PRECICE_DEBUG("There  NOTHIGN");
+    PRECICE_DEBUG("There  NOTHING");
```
Comment (luzpaz) [inline src/com/SocketCommunication.cpp:915]: `There's NOTHING` perhaps?

File: src/m2n/M2N.hpp [hunk 2]
```diff
@@ -219,8 +219,8 @@ class M2N {
   // and their re-partitioning, which could also not be moved to some fixture as the M2Ns
   // are created through the configuration.
   // See e.g. "CplSchemeTests/ExplicitCouplingSchemeTests/testConfiguredSimpleExplicitCoupling"
-  // This flag gives a loophole. It is set to false for normal use and modfied in the
-  // respective tests through a fried decleration.
+  // This flag gives a loophole. It is set to false for normal use and modified in the
+  // respective tests through a forward declaration.
```
Comment (luzpaz) [inline src/m2n/M2N.hpp:223]: Please review

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 3]
```diff
@@ -603,7 +603,7 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
     CHKERRV(ierr);
     ierr = MatMult(_matrixA, rescalingCoeffs, oneInterpolant);
     CHKERRV(ierr); // get the output of g(x) = 1
-    // set values close to zero to exactly 0.0, s.t. PointwiseDevide does not to devision on these entries
+    // set values close to zero to exactly 0.0, s.t. PointwiseDevide doesn't do division on these entries
```
Comment (luzpaz) [inline src/mapping/PetRadialBasisFctMapping.hpp:606]: Please review

File: src/math/geometry.cpp [hunk 4]
```diff
@@ -120,7 +120,7 @@ ResultConstants segmentPlaneIntersection(
   double d = pointOnPlane.dot(planeNormal);
 
   // The segment is represented by:
-  // firstPointSegment + (secondPointSegment - firstPointSegement) * t (2)
+  // firstPointSegment + (secondPointSegment - firstPointSegment) * t (2)
```
Comment (luzpaz) [inline src/math/geometry.cpp:123]: commented out source typo. Please review nonetheless.

File: src/m2n/M2N.hpp (hunk 2 above)
Comment (DavidSCN) [reply src/m2n/M2N.hpp:223]:
```suggestion
  // respective tests through a friend declaration.
```

**Review by DavidSCN — COMMENTED** [review]
That's a very nice and useful contribution @luzpaz, thanks a lot. Just one occurrence was misinterpreted, which is probably easy to fix. Considering the ease of use, I guess it would be a good idea to automate such a spellchecking in the future.

Apart from that, may I ask you to add a changelog entry? This PR fixes some user-visible typos in the log and error messages which should be visible in the changelog.

Also, please format the code properly. You can simply run `tools/formatting/format-all` from the root directory to do so.

**Comment by luzpaz** [comment]
Thanks! Will make the rest of the changes soon

**Comment by luzpaz** [comment]
> Also, please format the code properly. You can simply run `tools/formatting/format-all` from the root directory to do so.

<sigh> looks like `clang-8` is necessary. I have `clang-13`. I don't have an option to install clang-8 from my package manager. Please advise?

**Comment by DavidSCN** [comment]
The formatting only affects the two log messages in the `ValidationMacros` you fixed. You can use 

```
clang-format-13 -i src/precice/impl/ValidationMacros.hpp
```

and commit the two occurrences you touched, this should do the job. In case it still fails, I can also fix the formatting later on.

**Comment by luzpaz** [comment]
Pushed the clang-format in 91c0b93 per your request @DavidSCN

**Comment by DavidSCN** [comment]
Works nicely, thanks. What about the changelog entry? You can simply add a new file in `docs/changelog` called `1120.md`

**Comment by luzpaz** [comment]
@DavidSCN Sorry for the hand-holding. Will 9569ecd suffice?

**Comment by DavidSCN** [comment]
Let's go with it, thanks a lot @luzpaz

**Pull request merged**